test/logflare/backends/user_monitoring_test.exs

```diff
@@ -0,0 +1,258 @@
+    end
+
+    test "are not routed to user's system source when not monitoring", %{
+      user: user,
+      source: source
+    } do
+      {:ok, user} = Users.update_user_allowed(user, %{system_monitoring: true})
+      Users.update_user_allowed(user, %{system_monitoring: false})
+      system_source = Sources.get_by(user_id: user.id, system_source_type: :logs)
+
+      assert capture_log(fn ->
+               Logger.info("user not monitoring", source_id: source.id)
+             end) =~ "user not monitoring"
+
+      refute Enum.any?(
+               Backends.list_recent_logs(system_source),
+               &match?(%{body: %{"event_message" => "user not monitoring"}}, &1)
+             )
+    end
+
+    setup do
+      :ok =
+        :logger.add_primary_filter(
+          :user_log_intercetor,
+          {&Logflare.Backends.UserMonitoring.log_interceptor/2, []}
+        )
+
+      on_exit(fn ->
+        :logger.remove_primary_filter(:user_log_intercetor)
+      end)
+    end
+  end
+
+  describe "metrics" do
+    setup do
+      insert(:plan)
+
+      user_1 = insert(:user)
+      user_1 = Users.preload_defaults(user_1)
+
+      backend_1 =
+        insert(:backend,
+          user: user_1,
+          type: :webhook,
+          config: %{url: "http://test.com"},
+          default_ingest?: false
+        )
+
+      user_2 = insert(:user)
+      user_2 = Users.preload_defaults(user_2)
+
+      backend_2 =
+        insert(:backend,
+          user: user_2,
+          type: :webhook,
+          config: %{url: "http://test.com"},
+          default_ingest?: false
+        )
+
+      {:ok, user_1: user_1, user_2: user_2, backend_1: backend_1, backend_2: backend_2}
+    end
+
+    test "are routed to user's system source when flag is true", %{
+      user_1: user,
+      backend_1: %{id: backend_id}
+    } do
+      user |> Users.update_user_allowed(%{system_monitoring: true})
+      metadata = %{backend_id: backend_id}
+
+      # main exporter's metric keep function returns false
+      refute Telemetry.keep_metric_function(metadata)
+
+      # user exporter keep user specific metrics, and only that
+
+      system_source =
+        Sources.get_by(user_id: user.id, system_source_type: :metrics)
+
+      start_supervised!({SourceSup, system_source})
+
+      :telemetry.execute([:logflare, :test, :generic_metric], %{value: 123})
+      :telemetry.execute([:logflare, :test, :user_specific], %{value: 456}, metadata)
+
+      user_exporter_metrics =
+        OtelMetricExporter.MetricStore.get_metrics(:"system.metrics-#{user.id}")
+
+      refute match?(
+               %{{:last_value, "logflare.test.generic_metric.value"} => _},
+               user_exporter_metrics
+             )
+
+      assert match?(
+               %{
+                 {:last_value, "logflare.test.user_specific.value"} => %{
+                   %{backend_id: ^backend_id} => 456
+                 }
+               },
+               user_exporter_metrics
+             )
```

Comment:
as below, would prefer a more e2e test where we assert that the callback is called and the metrics are ingested.
for the mocked telemetry we can maybe export the values at a much faster interval with a higher `:export_period`